File: BennuLib/Legacy/CustomErrors.vb
Module CustomErrors



    '===============================================================================
    ' MODULE OVERVIEW
    '===============================================================================
    ' SpritePocket error constants and utility functions.
    '===============================================================================

    ' Error messages are defined in the ./res/SpritePocket.res resource file and are
    ' related to these error constants.
    Public Const ERR_CANNOTOPENFILE As Integer = 101
    Public Const ERR_NOTANFPGFILE As Integer = 102
    Public Const ERR_UNSUPPORTEDVERSION As Integer = 103
    Public Const ERR_PALETTETRUNCATED As Integer = 104
    Public Const ERR_GAMMATRUNCATED As Integer = 105
    Public Const ERR_CPINFOTRUNCATED As Integer = 106
    Public Const ERR_ANIMATIONFLAGACTIVE As Integer = 107
    Public Const ERR_BITMAPDATATRUNCATED As Integer = 108
    ' Public Const ERR_OBJECTUNAVAILABLE As Long = 109
    Public Const ERR_NOTAMAPFILE As Integer = 110
    Public Const ERR_NOTAPALFILE As Integer = 111
    Public Const ERR_MAPINDEXOUTOFRANGE As Integer = 112
    Public Const ERR_DIFFERENTDEPTHS As Integer = 113
    Public Const ERR_INVALIDCODE As Integer = 114
    Public Const ERR_PALETTEREQUIRED As Integer = 115
    Public Const ERR_UNSUPPORTEDDEPTH As Integer = 116
    Public Const ERR_SETPALETTEINVALIDDEPTH As Integer = 117
    Public Const ERR_INVALIDCPID As Integer = 118
    Public Const ERR_INVALIDMAPWIDTH As Integer = 119
    Public Const ERR_INVALIDMAPHEIGHT As Integer = 120
    Public Const ERR_INVALIDCOLORINDEX As Integer = 121
    Public Const ERR_INSUFFICIENTARRAYELEMENTS As Integer = 122
    Public Const ERR_FILEDOESNOTCONTAINPALETTE As Integer = 123
    Public Const ERR_CANNOTCONVERTMAPINFPG As Integer = 124
    Public Const ERR_INVALIDDEPTH As Integer = 125
    Public Const ERR_NOTANFNTFILE As Integer = 126
    Public Const ERR_SERIALIZEDMAPINVALID As Integer = 127
    Public Const ERR_INVALIDCOMPRESSIONLEVEL As Integer = 128
    Public Const ERR_ZLIB As Integer = 129
    Public Const ERR_FILEWRITE As Integer = 130

    ' FreeImage error constants
    Public Const ERR_FREEIMAGE As Integer = 2000
    Public Const ERR_FREEIMAGE_NOTKNOWNFILE As Integer = 2001

    ' As opposed to errors, warnings do not represent a SpritePocket failure and don't
    ' need to be handled
    Public Const WARNING_DUPLICATEDCODE As Integer = 1001
    Public Const WARNING_UNSUPPORTEDOPERATION As Integer = 1002
    Public Const WARNING_CANNOTCREATEOBJECT As Integer = 1003
    Public Const WARNING_NOTSTANDARDDECODER As Integer = 1004

End Module

File: BennuLib.Sharp/Legacy/CustomErrors.cs
namespace BennuLib
{
	static class CustomErrors
	{



		//===============================================================================
		// MODULE OVERVIEW
		//===============================================================================
		// SpritePocket error constants and utility functions.
		//===============================================================================

		// Error messages are defined in the ./res/SpritePocket.res resource file and are
		// related to these error constants.
		public const int ERR_CANNOTOPENFILE = 101;
		public const int ERR_NOTANFPGFILE = 102;
		public const int ERR_UNSUPPORTEDVERSION = 103;
		public const int ERR_PALETTETRUNCATED = 104;
		public const int ERR_GAMMATRUNCATED = 105;
		public const int ERR_CPINFOTRUNCATED = 106;
		public const int ERR_ANIMATIONFLAGACTIVE = 107;
		public const int ERR_BITMAPDATATRUNCATED = 108;
		// Public Const ERR_OBJECTUNAVAILABLE As Long = 109
		public const int ERR_NOTAMAPFILE = 110;
		public const int ERR_NOTAPALFILE = 111;
		public const int ERR_MAPINDEXOUTOFRANGE = 112;
		public const int ERR_DIFFERENTDEPTHS = 113;
		public const int ERR_INVALIDCODE = 114;
		public const int ERR_PALETTEREQUIRED = 115;
		public const int ERR_UNSUPPORTEDDEPTH = 116;
		public const int ERR_SETPALETTEINVALIDDEPTH = 117;
		public const int ERR_INVALIDCPID = 118;
		public const int ERR_INVALIDMAPWIDTH = 119;
		public const int ERR_INVALIDMAPHEIGHT = 120;
		public const int ERR_INVALIDCOLORINDEX = 121;
		public const int ERR_INSUFFICIENTARRAYELEMENTS = 122;
		public const int ERR_FILEDOESNOTCONTAINPALETTE = 123;
		public const int ERR_CANNOTCONVERTMAPINFPG = 124;
		public const int ERR_INVALIDDEPTH = 125;
		public const int ERR_NOTANFNTFILE = 126;
		public const int ERR_SERIALIZEDMAPINVALID = 127;
		public const int ERR_INVALIDCOMPRESSIONLEVEL = 128;
		public const int ERR_ZLIB = 129;

		public const int ERR_FILEWRITE = 130;
		// FreeImage error constants
		public const int ERR_FREEIMAGE = 2000;

		public const int ERR_FREEIMAGE_NOTKNOWNFILE = 2001;
		// As opposed to errors, warnings do not represent a SpritePocket failure and don't
		// need to be handled
		public const int WARNING_DUPLICATEDCODE = 1001;
		public const int WARNING_UNSUPPORTEDOPERATION = 1002;
		public const int WARNING_CANNOTCREATEOBJECT = 1003;

		public const int WARNING_NOTSTANDARDDECODER = 1004;
	}
}
